### PR TITLE
Game Design Patterns -- A Book Review

### PR DESCRIPTION
**Pull request by chuckbergeron**
- [x] Add images (sketch them up using Paper on an iPad?)
- [x] Ask the author of the book to give it a read and provide feedback
- [x] Change title of the article which I added basically as a working title joke

- [ ] **Fix the shared square image!**

### CODE REVIEW

**Comment by chuckbergeron** [comment]
screencapture-127-0-0-1-4000-2015-07-21-game-programming-design-patterns-and-you-book-review-html-1437177780311

**Comment by pcreux** [comment]
Great post chuck! Could you just add links to each design pattern: **Event Queue**, **Service**, **Dirty Flag**, **Singleton**, **State**, **Observer** and **Prototype**. I would definitely click on each of them to get a sense of what they are / what they do.

**Comment by chuckbergeron** [comment]
Latest version of this blog post is here. Ready to go for Tuesday!

https://s3-us-west-2.amazonaws.com/brewhouse-io/blog/game-programming-article-frozen/Game+Design+Patterns+--+A+Book+Review.html

File: _posts/2015-07-21-game-programming-design-patterns-and-you-book-review.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+Over time, as many developers work on a codebase the software will typically suffer from some common problems: tightly coupled code and spaghetti code are recurrent topics amongst developers -- and create messes most will want to avoid.
+
+For the past ten years I’ve been coding professionally, I’ve neglected to spend more time learning the underlying theories behind programming. Always preferring the hungry route of digging into the practical parts of development, blazing ahead while building tools and apps. That was until I took an interest in creating video games, which is when Robert Nystrom’s [Game Programming Patterns](http://gameprogrammingpatterns.com) caught my attention.
+
+<!-- break -->
+
+If you're not familiar with the concept of Design Patterns, they loosely outline various architectural methods you could employ to solve common problems anyone working on software could encounter. One of the more notable collections of Design Patterns is the book [Design Patterns: Elements of Reusable Object-Oriented Software](http://www.amazon.ca/Design-Patterns-Elements-Reusable-Object-Oriented/dp/0201633612) by four authors, appropratiely dubbed the "Gang of Four".
```
Comment (garbles) [inline _posts/2015-07-21-game-programming-design-patterns-and-you-book-review.md]: > appropratiely
:neckbeard:

**Comment by garbles** [comment]
That was definitely a fun read. Nice work, brah. (^________________________^     )

**Comment by patdryburgh** [comment]
@chuckbergeron Here are some suggested edits. Mostly punctuation, a few phrasing suggestions.

## Suggested Edits

As multiple developers work on a codebase over time, the software will typically suffer from some common problems:

“recurrent” to “recurring”

Nystrom has compiled almost twenty… - how many, exactly? “almost twenty” sounds weird.

web development, which I know rather well and video-game development, which I know nothing about

For instance, reading about the Double Buffer pattern and how it’s used when rendering graphics brought to mind the typical app deployment strategy used by Capistrano

The last step when completing a new deploy is to essentially flick a switch and direct the current symlink from the old, outdated directory to the brand new release directory.

add link to “graphics tearing”

which the Buffer pattern solves eloquently (I'm assuming you want this to match the section heading)

You’ve been given the task to create a spreadsheet application and you’re working on the Undo / Redo functionality for this spreadsheet. Or you’re building an turn-based strategy game and you need an Undo / Redo feature to allow the player to step back in time.

I imagine you can already see the correlation between the application world and video game world here.

Receiving commands from a user &mdash; whether to command units to move around a game board or entering a formula into a spreadsheet &mdash; is a fantastic feature of computers (remember pencils & erasers?).

You’re playing a game and you just found the secret spaceship locale, or killed your billionth goblin (or both, at the same time). A ding sounds and a message appears to let you know you’ve unlocked one of those nifty coveted trophies (trophies because I’m a PlayStation fan. Apologies, achievers).

If a message of interest is heard by the observer &mdash; say, a bit of code responsible for physics broadcasts a message when the player walks into that secret area &mdash; then the observer responds by providing the player with the trophy.

However, it was removed along with page and action caching – you can still use and install it from here.

I’m hoping this has inspired you to dig deeper into programming design patterns and instilled a deep and loving desire to learn all you can about the world of beautiful software architecture.

**Comment by chuckbergeron** [comment]
Fantastic feedback, thanks for that @patdryburgh !

**Pull request merged**